memo_app.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+require 'sinatra'
+require 'sinatra/reloader'
+require 'cgi'
+require 'json'
+require 'securerandom'
+
+MEMOS_FILE = 'memos.json'
+
+not_found do
+  'ページが見つかりません!'
+end
+
+def load_memo
+  JSON.parse(File.read(MEMOS_FILE), symbolize_names: true)
+end
+
+def add_memo(params)
+  existing_data = load_memo
+  existing_data[SecureRandom.uuid] = { title: params[:title], content: params[:content], created_at: Time.now, updated_at: Time.now }
+  save_memo(existing_data)
+end
+
+def update_memo(params, memo, existing_data)
+  memo.merge!(title: params[:title], content: params[:content], updated_at: Time.now)
+  save_memo(existing_data)
+end
+
+def delete_memo(params)
+  existing_data = load_memo
+  existing_data.delete(params[:id].to_sym)
```

Comment:
```suggestion
def delete_memo(id)
  existing_data = load_memo
  existing_data.delete(id)
```

使うのがidだけなら引数はそれだけにした方がいいですね